src/runner_manager.py

```diff
@@ -507,6 +508,7 @@ def reconcile(self, quantity: int, resources: VirtualMachineResources) -> int:
 
         if self.config.charm_state.is_metrics_logging_available:
             metric_stats = self._issue_runner_metrics()
+            runner_logs.remove_outdated_crashed()
```

Comment:
There is something that I didn't follow.
- A LXD instance will be available even if the runner crashed
- If log is available, it will check for old crashed files and remove the ones older than 7 days
- Then for unhealthy runners it will check for logs to be pulled

How do you know if you are not getting the ones that were already pulled?
And if the runners crashed but the instance will be still available, couldn't you remove it after being successfully pulled?

Reply:
BTW What if log is disabled, the old files would be left there right?

Reply:
>How do you know if you are not getting the ones that were already pulled?
And if the runners crashed but the instance will be still available, couldn't you remove it after being successfully pulled?

The unhealthy instances are removed after the logs are pulled by calling `runner.remove`.

Reply:
> BTW What if log is disabled, the old files would be left there right?

This is a good point, I have updated the code to always remove outdated logs.